src/serial.rs

```diff
@@ -320,21 +319,19 @@ where
         assert!(brr >= 16, "impossible baud rate");
         self.usart.brr.write(|w| unsafe { w.bits(brr) });
 
-        let (parity_control_enable, parity) = match config.parity {
+        // Configure word
+        let (parity_is_used, parity_is_odd) = match config.parity {
             Parity::ParityNone => (false, false),
             Parity::ParityEven => (true, false),
             Parity::ParityOdd => (true, true),
         };
-        self.usart.cr1.modify(|_r, w| {
-            w.m()
-                .bit(match config.wordlength {
-                    WordLength::DataBits8 => false,
-                    WordLength::DataBits9 => true,
-                })
-                .ps()
-                .bit(parity)
-                .pce()
-                .bit(parity_control_enable)
+        self.usart.cr1.modify(|_r, w| { w
+            .m().bit(match config.wordlength {
+                WordLength::DataBits8 => false,
+                WordLength::DataBits9 => true,
+            })
+            .ps().bit(parity_is_odd)
+            .pce().bit(parity_is_used)
         });
```

Comment:
If you want to write it in 1 line you can use this way
```suggestion
        self.usart.cr1.modify(|_r, w| {
            w.m().bit(match config.wordlength {
                WordLength::DataBits8 => false,
                WordLength::DataBits9 => true,
            });
            w.ps().bit(parity_is_odd);
            w.pce().bit(parity_is_used)
        });
```
or rustfmt

Reply:
Thanks for the hint